mllib/src/main/scala/org/apache/spark/ml/regression/RandomForestRegressor.scala

```diff
@@ -168,15 +173,37 @@ class RandomForestRegressionModel private[ml] (
   // Note: We may add support for weights (based on tree performance) later on.
   private lazy val _treeWeights: Array[Double] = Array.fill[Double](_trees.length)(1.0)
 
+  @Since("2.1.0")
+  /** @group getParam */
+  def setVarianceCol(value: String): this.type = set(varianceCol, value)
+
   @Since("1.4.0")
   override def treeWeights: Array[Double] = _treeWeights
 
+  private def predictVariance(features: Vector): Double = {
```

Comment:
ping @jkbradley to double check the formula in the academic citation is widely used and accepted.